src/main/java/nl/stil4m/mollie/concepts/AbstractConcept.java

```diff
@@ -0,0 +1,79 @@
+package nl.stil4m.mollie.concepts;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import nl.stil4m.mollie.RequestExecutor;
+import nl.stil4m.mollie.ResponseOrError;
+import nl.stil4m.mollie.domain.Page;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+
+import java.io.IOException;
+import java.nio.charset.UnsupportedCharsetException;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class AbstractConcept<T> implements Concept<T> {
+    private static final Collector<CharSequence, ?, String> URL_JOINER = Collectors.joining("/");
```

Comment:
it's the java Collectors.joining return type; casting is even more ugly, inlining it might be better